### PR TITLE
feat: staking

### PR DESCRIPTION
**Pull request by maxwellmattryan**
# Description of change
This PR includes the implementation for the staking feature, where users can stake their IOTA tokens for [Assembly (ASMB)](https://assembly.sc) and [Shimmer (SMR)](https://shimmer.network) airdrops. This will be done in the new dashboard accessible by the new sidebar icon.

## Changes
- Adds staking dashboard containing new dashboard tiles: `StakingSummary`, `StakingAirdrop`, and `StakingInfo`
- Adds popup flow allowing users with either a software or Ledger profile to stake or unstake their individual wallets (handles the case of partially staked accounts*)
- Adds tooltips to the `AccountTile` component displaying a warning if the account has either not reached the minimum airdrop (and is staked) or if their are partially staked funds on an account*
- Adds warning popup if users try to send from an account that is currently staked (the result is that __some or all__ of the outputs on that account will be spent, meaning the account may or may not still be staked after the regular transfer)

\* Partially staked funds refer to when an account has 1+ outputs with the `PARTICIPATE` index inside of an embedded indexation payload (in other words are being tracked by the "Participation" node plugin) __and__ 1+ outputs that __do NOT__ have this type of embedded indexation payload.   

## Links to any relevant issues
None

## Type of change
- New 
 
## How the changes has been tested
Tested on:
- MacOS (Catalina 10.15.7)
- Windows 10

## Change checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by nicole-obrien** [comment]
💪🏻🥩

File: packages/desktop/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "desktop",
-    "productName": "Firefly",
-    "version": "1.2.0",
+    "productName": "Firefly Beta",
+    "version": "1.3.0-beta",
```
Comment (nicole-obrien) [inline packages/desktop/package.json:4]: Change before merge:
```suggestion
    "productName": "Firefly",
    "version": "1.3.0",
```

File: packages/desktop/package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
         "webpack-dev-server": "^3.11.2"
     },
     "build": {
-        "productName": "Firefly",
+        "productName": "Firefly Beta",
```
Comment (nicole-obrien) [inline packages/desktop/package.json:51]: Revert back before merge:
```suggestion
        "productName": "Firefly",
```

File: packages/desktop/scripts/notarize.macos.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ exports.default = async () => {
 
     await notarize({
         appBundleId: 'org.iota.firefly',
-        appPath: path.resolve(__dirname, '../out/mac/Firefly.app'),
+        appPath: path.resolve(__dirname, '../out/mac/Firefly Beta.app'),
```
Comment (nicole-obrien) [inline packages/desktop/scripts/notarize.macos.js:22]: Revert back before merge:
```suggestion
        appPath: path.resolve(__dirname, '../out/mac/Firefly.app'),
```

**Review by nicole-obrien — COMMENTED** [review]
Highlighting some stuff that needs reverting/changing before release.

File: packages/shared/locales/en.json [hunk 4]
```diff
@@ -675,6 +731,40 @@
             "accountName": "Account name:",
             "typePassword": "Type your password to allow exporting"
 
+        },
+        "stakingNotice": {
+            "upcomingDate": "Staking commences Dec 1st",
```
Comment (maxwellmattryan) [inline packages/shared/locales/en.json:736]: ⚠️ Needs updating before merge

**Comment by maxwellmattryan** [comment]
> Highlighting some stuff that needs reverting/changing before release.

Sweet 🙏

**Review by cvarley100 — APPROVED** [review]

**Pull request merged**